### DISCUSSION TITLE
Product Category: Infinite Scroll

### DISCUSSION THREAD

**Discussion by Mj6078** [community/community#145235] [Programming Help] [answered]
### Body

I am trying to make my category page infinite scroll to go till the end of the product and I have increased the product limit in JSON as well as

 ---

category:

  products:

limit: {{theme_settings.product-count}}

  shop_by_price: true

--- 

Here, but it is stuck at the 100 Max Products. Does anyone know how to make an infinite scroll in the peak big commerce theme? Additionally, I also tried using this [Medium article](https://medium.com/bigcommerce-developer-blog/how-to-add-infinite-scroll-to-category-pages-6c991750a8d5), which is available online, but I still am not able to make an infinite scroll. Can anyone face this same problem and have a solution that will be very appreciated? Thank you...!!!!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by denis23x** [comment]
Hi @Mj6078 

I think you should increment the page and concat the answer you receive to the existing list, this will be a correct implementation, increasing the limit is not correct.

**Reply by Mj6078** [reply]
Thank you For your replay But my mind is not over come on this limit line 
limit: {{theme_settings.product-count}} If I put here 50, then it is showing 50, and if I put any value that is below 100, then it is working perfect but not going above, so I guess somewhere in the code if I can change that value, then that is the easy way for me as well as for my client in the future, so do you know anything about this in big commerce? How does this work?

**Reply by denis23x** [reply, marked as answer]
Yes, I understand you completely, but as I said before, this is the wrong approach ⛔️

What if you put 1000 or 100000 there? Imagine how huge the request will be, you will inevitably encounter other errors.

I found a couple topics, please check 

- https://support.bigcommerce.com/s/question/0D54O000069iOrwSAE/is-it-possible-to-increase-the-maximum-number-of-product-per-page-on-a-blueprint-theme-past-100in-store-setup-store-settings-display-i-have-found-an-option-to-increase-the-number-up-to-100-but-am-not-allowed-to-make-it-greater-than-100?language=en_US
- https://support.bigcommerce.com/s/question/0D51B00004NUIFFSA5/is-there-a-way-to-get-around-the-100-products-per-page-limit?language=en_US

You need to do a real infinite scroll, by detecting the bottom of the page with JavaScript, and request via API the next page (current page + 1), received response you should concat to the original list, and so on.

Increasing the limit (even if it is somehow possible) will only slow down the initial page load (at least, at most it will lead to new errors), it is not infinite scrolling in itself in any case.

**Reply by Mj6078** [reply]
That's true; that will increase the load on that page, so I should go for an infinite scroll. I am new at this. Thank you for sharing this thing; it is really helpful for me. Thank you for your help.
